### ISSUE TITLE
Comparators error when using pyspark

### ISSUE THREAD

**Issue by hbashary** [G-Research/spark-extension#225]
Trying to run example in documentation using pyspark but keep getting the following error - 
**AttributeError: 'DiffOptions' object has no attribute 'withComparator' .** 

Running this in a Glue notebook with Spark version 3.3 and spark-extension_2.12-2.8.0. Same issue when upgrading to spark-extension_2.13-2.11.0.  Is this method supported for the python api?

**Create 2 dataframes** 

    df_1 = spark.createDataFrame([
        Row(id=1, value=1.0),
        Row(id=2, value=2.0),
        Row(id=3, value=3.0),
    ])

    df_2 = spark.createDataFrame([
        Row(id=1, value=1.0),
        Row(id=2, value=2.02),
        Row(id=3, value=3.05),
    ])

**Run Comparator method**

    from pyspark.sql.types import DoubleType
    from gresearch.spark.diff import DiffOptions, DiffMode, DiffComparators
    
    options = DiffOptions().with_change_column("changes")\
                           .withComparator(DiffComparators.epsilon(0.01).asRelative().asInclusive(), DoubleType)
    
    df_1.diff_with_options(df_2, options, "id").show()

**Error - AttributeError: 'DiffOptions' object has no attribute 'withComparator'**

**Comment by hbashary** [comment]
Thanks for the quick response. One last question - the map attribute doesn't seem to be supported for python. 

options = DiffOptions().with_change_column("changes")\\
                                       .with_data_type_comparator(DiffComparators.map[K,V](false))

Error - AttributeError: type object 'DiffComparators' has no attribute 'map'

**Comment by hbashary** [comment]
options = DiffOptions().with_change_column("changes")\\
                                       .with_data_type_comparator(DiffComparators.map[K,V](false))

Error - AttributeError: type object 'DiffComparators' has no attribute 'map'

**Comment by EnricoMi** [comment]
Right, the Python API does not support the Map comparator. I haven't yet figured out how to get the key and value types `K` and `V` from Python to Scala.

**Comment by hbashary** [comment]
Thanks Enrico.

**Comment by EnricoMi** [comment]
I have found a way to provide the `MapDiffComparator` to Python API: #226

**Comment by EnricoMi** [comment]
This has been released.
